### ISSUE TITLE
missing data in read.fstat()

### ISSUE THREAD

**Issue by rkartzin** [thibautjombart/adegenet#88]
Hello, I am running version 2.0.1 and cannot read in my fstat files correctly using read.fstat(). The fstat file has 3 characters per allele (e.g. "120130") with missing data as "0". Adegenet reads the missing data as an allele so I have extra alleles per locus. 
```
> 	dat<-adegenet::read.fstat(filename)

 Converting data from a FSTAT .dat file to a genind object... 


...done.

> dat
/// GENIND OBJECT /////////

 // 143 individuals; 12,348 loci; 37,044 alleles; size: 27.2 Mb

 // Basic content
   @tab:  143 x 37044 matrix of allele counts
   @loc.n.all: number of alleles per locus (range: 3-3)
   @loc.fac: locus factor for the 37044 columns of @tab
   @all.names: list of allele names for each locus
   @ploidy: ploidy of each individual  (range: 2-2)
   @type:  codom
   @call: adegenet::read.fstat(file = filename)

 // Optional content
   @pop: population of each individual (group size range: 23-24)
> head(dat@all.names)
$`1`
[1] "120" "0"   "110"

$`4`
[1] "120" "0"   "100"

$`15`
[1] "120" "0"   "100"

$`23`
[1] "130" "0"   "110"

$`24`
[1] "100" "120" "0"  

$`30`
[1] "110" "0"   "130"
```
I think that maybe read.fstat() expects missing data to be encoded as "000" in this case. Is there a way around this, short of replacing all the missing data in my files with "000"? It seems like recording a missing genotype as "0" instead of "000000" is not uncommon - I have found that both write.fstat() in hierfstat and PGDspider do it that way.
Thank you!

**Comment by thibautjombart** [comment]
Hi there, 
thanks for the report. Can you provide a toy (reproducible) example so that we can work out a patch for it? I think `df2genind` only allows a single code for NAs, but we could add this as an argument to `read.fstat`.

**Comment by rkartzin** [comment]
Hello!
Toy dataset is attached. Thank you!

**Comment by thibautjombart** [comment]
Hi again, 
I could not load the data, but the issue should be fixed as of 9c569e145ec9af21988a26db521bd5f3fb5064d0
In the old definition of FSTAT missing data should be coded by the same number of characters as the other alleles, but now `read.fstat` will handle strings of 1 to 8 zeros as NA.
Can you confirm the fix?
